src/js/tracks/text-track.js

```diff
@@ -318,4 +102,245 @@ var indexOf = function(searchElement, fromIndex) {
   return -1;
 };
 
+/**
+ * A single text track as defined in:
+ * https://html.spec.whatwg.org/multipage/embedded-content.html#texttrack
+ *
+ * interface TextTrack : EventTarget {
+ *   readonly attribute TextTrackKind kind;
+ *   readonly attribute DOMString label;
+ *   readonly attribute DOMString language;
+ *
+ *   readonly attribute DOMString id;
+ *   readonly attribute DOMString inBandMetadataTrackDispatchType;
+ *
+ *   attribute TextTrackMode mode;
+ *
+ *   readonly attribute TextTrackCueList? cues;
+ *   readonly attribute TextTrackCueList? activeCues;
+ *
+ *   void addCue(TextTrackCue cue);
+ *   void removeCue(TextTrackCue cue);
+ *
+ *   attribute EventHandler oncuechange;
+ * };
+ *
+ * @param {Object=} options Object of option names and values
+ * @extends EventTarget
+ * @class TextTrack
+ */
+class TextTrack extends EventTarget {
+  constructor(options = {}) {
+    super();
+    if (!options.tech) {
+      throw new Error('A tech was not provided.');
+    }
+
+    let tt = this;
+
+    if (browser.IS_IE8) {
+      tt = document.createElement('custom');
+
+      for (let prop in TextTrack.prototype) {
+        if (prop !== 'constructor') {
+          tt[prop] = TextTrack.prototype[prop];
+        }
+      }
+    }
+
+    tt.tech_ = options.tech;
+
+    let mode = TextTrackEnum.TextTrackMode[options.mode] || 'disabled';
+    let kind = TextTrackEnum.TextTrackKind[options.kind] || 'subtitles';
+    let label = options.label || '';
+    let language = options.language || options.srclang || '';
+    let id = options.id || 'vjs_text_track_' + Guid.newGUID();
+
+    if (kind === 'metadata' || kind === 'chapters') {
+      mode = 'hidden';
+    }
+
+    tt.cues_ = [];
+    tt.activeCues_ = [];
+
+    let cues = new TextTrackCueList(tt.cues_);
+    let activeCues = new TextTrackCueList(tt.activeCues_);
+    let changed = false;
+    let timeupdateHandler = Fn.bind(tt, function() {
+      this.activeCues;
+      if (changed) {
+        this.trigger('cuechange');
+        changed = false;
+      }
+    });
+
+    if (mode !== 'disabled') {
+      tt.tech_.on('timeupdate', timeupdateHandler);
+    }
+
+    Object.defineProperty(tt, 'kind', {
+      get() {
+        return kind;
+      },
+      set: Function.prototype
```

Comment:
Since we're already in here, can we change `Function.prototype` to `() => {}` or `const noop = () => {}` and then use `noop`. It was fun while it lasted.